kid0012.md

```diff
@@ -0,0 +1,62 @@
+# KID0012 - Super Watchers
+
+## Overview
+
+This document describes a feature of the KERI ecosystem that has been referred to as "super watchers".
+
+Watchers are already implemented; essentially, they are witnesses that do not directly record KEL and TEL events, but simply observe those events as recorded by other witnesses. Watchers typically observe the state of many AIDs (as reported by the AIDs' associated witnesses). Since we expect delegation and credential issuance and revocation events to be anchored to the KEL, watchers also observe these events. They do this to provide a consolidated and curated view. Watchers are a convenient way to centralize how state is *queried*, without actually centralizing how that state is *proved* or *trusted*. Because watchers observe events in many places, they are also an important mechanism for detecting and responding to duplicity.
+
+An ordinary watcher has an enumeration of AIDs for which it observes events, and it may have a small set of stakeholders that depend upon it. A "super watcher" is a watcher with a much larger scope of concern. It is typically intended as a clearinghouse for information throughout an ecosystem, and it is configured to be greedy -- having a dynamic scope that continually expands. For example, a watcher that attempts to watch all AIDs that issue or receive vLEIs — automatically adding new AIDs and their associated witnesses to its scope whenever new vLEIs are encountered — might be a super watcher.
+
+The KERI whitepaper laid out some additional vision about watchers. A "juror" is a specialized watcher that doesn't just passively note things; it actively reports any evidence of duplicity that it detects. And a "judge" is a watcher that goes one step further -- it is configured with policies that decide how to resolve the duplicity.
+
+Super watchers should almost certainly be jurors, and some of them may want to be judges, as well.
+
+We want super watchers to operate efficiently and interoperably in any KERI-oriented ecosystem, because they guarantee the rapid and accurate flow of information. We imagine that organizations with a big stake in an ecosystem's health will operate super watchers as a public good, in exchange for the benefits to their own reputation and their own customer experience.
+
+## Desired End State
+
+When the design described by this KID is implemented, the following things should be true about an ecosystem that has deployed super watchers:
+
+1. It is practical for super watchers to watch millions of AIDs. Assuming modest rates of change in identifier state, and setting aside linear growth in storage requirements, this scale does not impose undue burden on witnesses or watchers, insofar as network chattiness, RAM, and other resources is concerned.
+2. Super watchers can discover and watch all AIDs that may ever be serviced by particular witnesses, not just AIDs that have been explicitly enumerated to the super watchers.
```

Comment:
In the whitepaper I always assumed that AIDs, controllers, and witnesses may want to enforce their own privacy controls within their given deployment.  Should a watcher be able to discover and watch AIDs that haven't been been enumerated to the watcher by the witness?  It seems like this would only be a detriment to the controller's verification in networks that it hasn't propagated to that rely on these super watchers but that it might be desirable for controller networks that don't want to rely/allow discovery of arbitrary aids (like for adversaries who might want to enumerate all potential aids within an arbitrary witness network).  Obviously, the witnesses could always implement this anyways.

Maybe: "super watchers can discover and watch all AIDs that are made discoverable a given witness".

Reply:
Good improvement! I'll commit a fix along these lines.